pg_search/src/postgres/customscan/pdbscan/exec_methods/fast_fields/mod.rs

```diff
@@ -266,8 +275,19 @@ pub unsafe fn pullup_fast_fields(
 
         if let Some(var) = nodecast!(Var, T_Var, (*te).expr) {
             if (*var).varno as i32 != rti as i32 {
-                // this TargetEntry's Var isn't from the same RangeTable as we were asked to inspect,
-                // so just skip it
+                // We expect all Vars in the target list to be from the same range table as the
+                // index we're searching, so if we see a Var from a different range table, we skip it.
+                if is_execution_time {
+                    // This is a sanity check to ensure that the target list is consistent with the
+                    // index we're searching. As we're not supporting JOINs and Projection, at
+                    // execution time (not planning time), we expect all Vars in the target list to
+                    // be from the same range table as the index we're searching.
+                    pgrx::warning!(
+                        "Encountered a Var with a different range table index: {} (expected {})",
+                        (*var).varno,
+                        rti
+                    );
```

Comment:
If this warning triggers, then we're about to panic due to the assertion in `compute_exec_which_fast_fields`: I could imagine that it is better to ensure that these are all logged before we panic, but should maybe indicate that here.

But also: we already have a facility for handling failures here: the `collect_fast_field_try_for_attno` function immediately below returns `None`. So maybe remove all `continue`s, and then we could guarantee that this function either returns a list of the correct length, or `None`.